app/scripts/models/account.js

```diff
@@ -462,23 +462,29 @@ define(function (require, exports, module) {
      *   @param {String} [options.resume] - Resume token to send
      *   in verification email if user is unverified.
      *   @param {String} [options.unblockCode] - Unblock code.
+     *   @param {String} [options.originalLoginEmail] - Login used to login with originally.
      * @returns {Promise} - resolves when complete
      */
     signIn (password, relier, options = {}) {
+      var email = this.get('email');
       return p().then(() => {
-        var email = this.get('email');
         var sessionToken = this.get('sessionToken');
-
         if (password) {
-          return this._fxaClient.signIn(email, password, relier, {
+          const signinOptions = {
             metricsContext: this._metrics.getFlowEventMetadata(),
             reason: options.reason || SignInReasons.SIGN_IN,
             resume: options.resume,
             // if the email case is incorrect, handle it locally so the model
             // can be updated with the correct case.
             skipCaseError: true,
             unblockCode: options.unblockCode
-          });
+          };
+
+          if (options.originalLoginEmail) {
```

Comment:
Can you leave a short explanation of how this comes to be populated/how the flow works? I had to read the code several times to come to a basic understanding. What does the auth-server do with `originalLoginEmail`? What if `originalLoginEmail` itself is incorrectly cased?